campos_potenciais2.py

```
@@ -0,0 +1,278 @@
+
+    # primeiro: tentamos usar getObjectMatrix do sensor e do robô
+    try:
+        sensor_mat = sim.getObjectMatrix(sensorHandle)  
+        robot_mat = sim.getObjectMatrix(robotHandle)
+
+        robot_mat_inv = sim.getMatrixInverse(robot_mat)
+        rel = sim.multiplyMatrices(robot_mat_inv, sensor_mat)  # 3x4 matrix
+        obs_in_robot_3 = sim.multiplyVector(rel, detectedPoint)
+        obs_in_robot = np.array(obs_in_robot_3)
+        dist = np.linalg.norm(obs_in_robot)
+        return True, obs_in_robot, dist
```

Ruff (0.13.3)

Consider moving this statement to an `else` block

(TRY300)